examples/plotting/zoom_map_example.py

```diff
@@ -0,0 +1,62 @@
+"""
+===================================
+zoom-out a particular region of map
+===================================
+How to plot a zoom-out image using matplotlib.
+"""
+import matplotlib.pyplot as plt
+from matplotlib.patches import ConnectionPatch
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import AIA_171_IMAGE
+
+###############################################################################
+# Define a region of interest.
+length = 500 * u.arcsec
+x0 = -50 * u.arcsec
+y0 = -30 * u.arcsec
+
+###############################################################################
+# create a sunpy Map and submap over the region of interest.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+bottom_left = SkyCoord(x0 - length, y0-length,
+                       frame=smap.coordinate_frame)
+top_right = SkyCoord(x0 + length, y0+length,
+                     frame=smap.coordinate_frame)
+submap = smap.submap(bottom_left, top_right=top_right)
+
+fig = plt.figure(figsize=(16, 5))
+ax1 = fig.add_subplot(1, 2, 1, projection=smap)
+smap.plot(axes=ax1, clip_interval=(1, 99.9)*u.percent)
+
+# Draw a box on the image
+smap.draw_rectangle(bottom_left, color='k', height=length * 2, width=length * 2)
+
+# Create a second axis on the plot.
+ax2 = fig.add_subplot(1, 2, 2, projection=submap)
+
+submap.plot()
+
+# Add a overlay grid.
+submap.draw_grid(grid_spacing=20*u.deg)
+
+# name the zoomed plot.
+ax2.set_title('Zoomed View', fontsize=8)
+
+# Add connectors
+xpix, ypix = smap.world_to_pixel(top_right)
+con1 = ConnectionPatch(
+    (0, 1), (xpix.value, ypix.value), 'axes fraction', 'data', axesA=ax2, axesB=ax1,
+    arrowstyle='-', color='k', lw=1
+)
+xpix, ypix =smap.world_to_pixel(SkyCoord(top_right.Tx, bottom_left.Ty, frame=smap.coordinate_frame))
```

Comment:
```suggestion
xpix, ypix = smap.world_to_pixel(SkyCoord(top_right.Tx, bottom_left.Ty, frame=smap.coordinate_frame))
```

Reply:
Hey @nabobalis thanks for the reviews. Could you please look at this [comment](https://github.com/sunpy/sunpy/pull/4920#discussion_r567300217) from @wtbarnes
Maybe this will give us a clear insight of what has to be done. I am going with Will's wishes it seems : )
Wow isn't Sun so beautiful with it's flares.


![zoomed](https://user-images.githubusercontent.com/72349558/107850420-56445c80-6e28-11eb-9ad3-2c18a5672250.png)